lib/collection/src/operations/types.rs

```diff
@@ -71,6 +71,33 @@ pub enum CollectionStatus {
     Red,
 }
 
+/// Current state of the shard (supports same states as the collection)
+/// `Green` - all good. `Yellow` - optimization is running, `Red` - some operations failed and was not recovered
+#[derive(Debug, Serialize, JsonSchema, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum ShardStatus {
```

Comment:
Do I get it right, that you don't want to reuse CollectionStatus, cause we don't want to change names exposed to the clients? (make sense, btw)

Reply:
Yeah that plus it makes more semantic sense to call it `ShardStatus` than `CollectionStatus`

Reply:
Isn't it only in telemetry? I believe we don't really care about changes there, because we don't guarantee stable typing there.

Reply:
It's still better semantics to call it ShardStatus?

Reply:
> It's still better semantics to call it ShardStatus...

IMO, yes, `ShardStatus` is better.